packages/core/src/hooks/useCallResilency.test.tsx

```diff
@@ -199,6 +202,51 @@ describe('useCall Resilency tests', () => {
             expect(result.current.secondChainBlockNumber).to.be.equal(2)
             expect(result.current.chainId).to.be.equal(1337)
           })
+
+          it('Does not do duplicate polls for data', async () => {
+            const { result, waitForCurrent, rerender } = await renderDAppHook(
+              () => {
+                const { chainId, error } = useEthers()
+                const { blockNumber: firstChainBlockNumber } = useBlockMeta({ chainId: 1337 })
+                return { chainId, firstChainBlockNumber, error }
+              },
+              {
+                config: {
+                  readOnlyChainId: config.readOnlyChainId,
+                  readOnlyUrls: {
+                    [1337]: config.readOnlyUrls![1337],
+                  },
+                  multicallAddresses: {
+                    [1337]: config.multicallAddresses![1337],
+                  },
+                  pollingInterval: 500,
+                },
+              }
+            )
+            await waitForCurrent((val) => val.chainId !== undefined && val.firstChainBlockNumber !== undefined)
+            expect(result.error).to.be.undefined
+
+            const calls: string[] = []
+
+            const originalCall: providers.StaticJsonRpcProvider['call'] = (config.readOnlyUrls![1337] as any).call
+            ;(config.readOnlyUrls![1337] as any).call = async function (...args: any[]): Promise<any> {
+              if (args[1] === 2) {
+                // In this test, let's take a look at calls made for blockNumber 2.
+                calls.push(JSON.stringify(args))
+              }
+              return await originalCall.apply(config.readOnlyUrls![1337], args as any)
+            }
+
+            await miners[0].sendTransaction({ to: constants.AddressZero })
+            rerender()
```

Comment:
Is it necessary?

Reply:
Somehow, yes - without the fix, the test is not red without `rerender`